### ISSUE TITLE
Show a message if the selected camera does not match the resolution

### ISSUE THREAD

**Issue by felixzapata** [livekit/client-sdk-js#582]
Hi, I have a USB camera with an available size of 640x480. This camera appears on the devices list but when it is selected, the preview uses my FaceTime HD camera. The default size for cameras in this example is 1280x720 I think.

Is it possible to get an `OverconstrainedError` in order to inform a user the selected device is not available for use?

**Comment by lukasIO** [comment]
Had a look at this again.
Generally when using the react components for selecting a device, we will not pass in the `exact` parameter. 
I will take a look at adding this to the react package, but also check out https://github.com/livekit/components-js which includes the new package for react.

**Comment by felixzapata** [comment]
But I also have the same problem if I try to use the Web (JS/TS) version. Besides, in my final development, I am not using React.

When I use the API to create local tracks for video and microphone using createLocalVideoTrack, createLocalAudioTrack, or createLocalTracks, I never see the preview of my VGA camera selected on the select box unless I pass a preset to the method:

```
const track = await createLocalVideoTrack({
    deviceId: devicesIds.get('videoinput'),
    resolution: { width: VideoPresets.h540.width, height: VideoPresets.h540.height },
  });
```

Because there is no message or something in the console, the user will never know why a camera listed in the select box is not working.

**Comment by lukasIO** [comment]
try with 
```
const track = await createLocalVideoTrack({
    deviceId: {exact: devicesIds.get('videoinput')},
    resolution: { width: VideoPresets.h540.width, height: VideoPresets.h540.height },
  });
  ```

**Comment by felixzapata** [comment]
hi, it works the same as my previous code:

```
const track = await createLocalVideoTrack({
    deviceId: devicesIds.get('videoinput'),
    resolution: { width: VideoPresets.h540.width, height: VideoPresets.h540.height },
  });
```

Both cameras now are connected with a 960x540 resolution. So, I assume we lose the possibility to use the HD camera with a bigger resolution.

By the way, I've tested the VGA camera alone with a mac mini in a firefox browser and after the browser request the device's permission, I was unable to view my preview.

**Comment by lukasIO** [comment]
ah sorry, I meant for you to try 
```
const track = await createLocalVideoTrack({
    deviceId: {exact: devicesIds.get('videoinput')}
  });
```

the difference being that you provide `exact` for the device, so the resolution shouldn't take priority any more!

**Comment by felixzapata** [comment]
ok, that works for me. Thanks.

**Comment by lukasIO** [comment]
you're right, I opened https://github.com/livekit/client-sdk-js/pull/612 to address this
